presentation/src/main/java/com/knocklock/presentation/lockscreen/mapper/NotificationMapper.kt

```diff
@@ -0,0 +1,157 @@
+package com.knocklock.presentation.lockscreen.mapper
+
+import android.app.PendingIntent
+import android.content.pm.ApplicationInfo
+import android.content.pm.PackageManager
+import android.graphics.drawable.Drawable
+import android.os.Build
+import android.service.notification.StatusBarNotification
+import com.knocklock.presentation.lockscreen.model.Notification
+import com.knocklock.presentation.util.TimeWithMeridiem
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import com.knocklock.domain.model.Notification as NotificationModel
+
+/**
+ * @Created by 김현국 2023/03/07
+ */
+
+/**
+ * StatusBarNotification를 Array<[NotificationModel]>로 변환합니다.
+ *
+ * StatusBarNotification를 수정할 수 없기 때문에 List->Array로 변환합니다.
+ *
+ * @param statusBarNotifications Array<out [StatusBarNotification]> activeNotifications 입니다.
+ * @return Array<[NotificationModel]>
+ */
+fun toModel(statusBarNotifications: Array<out StatusBarNotification>, packageManager: PackageManager): Array<NotificationModel> {
+    return statusBarNotifications.asSequence().filter { it.toModel(packageManager) != null }.map { it.toModel(packageManager)!! }.toList().toTypedArray()
+}
+
+/**
+ * StatusBarNotification에서 Database에 저장할 키를 반환합니다.
+ *
+ * StatusBarNotification에서 title이나 content가 있는 경우 데이터베아스의 저장할 키를 생성합니다.
+ * @param packageManager AppTitle을 가져오기 위한 packageManager입니다.
+ * @return null OR Group.Key
+ */
+fun StatusBarNotification.getDatabaseKey(packageManager: PackageManager): String? {
+    val title: String = convertString(this.notification.extras.getCharSequence("android.title"))
+    val content: String = convertString(this.notification.extras.getCharSequence("android.text"))
+    return if (title != "" || content != "") {
+        val appTitle = getDrawableAndAppTitle(packageManager, packageName).first ?: ""
+        val subText: String = convertString(this.notification.extras.getCharSequence("android.subText"))
+
+        if (subText == "")getGroupKey(packageName, appTitle, title) else getGroupKey(packageName, appTitle, subText)
```

Comment:
"" 비교는 isEmpty extensions을 쓰면 더 가독성이 좋을 것 같아요!

```suggestion
    return if (title.isNotEmpty() || content.isNotEmpty()) {
        val appTitle = getDrawableAndAppTitle(packageManager, packageName).first ?: ""
        val subText: String = convertString(this.notification.extras.getCharSequence("android.subText"))

        if (subText.isEmpty) getGroupKey(packageName, appTitle, title) else getGroupKey(packageName, appTitle, subText)
```

Reply:
동의합니다 ㅎㅎ

Reply:
@jaeryo2357 @KimHance [b4f68e1](https://github.com/Knock-Lock/Knock-Lock/pull/101/commits/b4f68e10bbc60762f74fede162e832aaa76f6b33) 수정했습니당